tool/checks/rules/no_solo_tests.dart

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) 2022, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:analyzer/dart/ast/ast.dart';
+import 'package:analyzer/dart/ast/visitor.dart';
+import 'package:linter/src/analyzer.dart';
+
+const _desc = r"Don't commit soloed tests.";
+
+const _details = r'''
+**DON'T** commit a soloed test.
+
+**BAD:**
+```dart
+@soloTest
+test_myTest() async {
+  ...
+}
+```
+
+**GOOD:**
+```dart
+test_myTest() async {
+  ...
+}
+```
+''';
+
+class NoSoloTests extends LintRule {
+  NoSoloTests()
+      : super(
+            name: 'no_solo_tests',
+            description: _desc,
+            details: _details,
+            group: Group.errors);
+
+  @override
+  void registerNodeProcessors(
+      NodeLintRegistry registry, LinterContext context) {
+    var visitor = _Visitor(this);
+    registry.addMethodDeclaration(this, visitor);
+  }
+}
+
+
+class _Visitor extends SimpleAstVisitor<void> {
+  final LintRule rule;
+
+  _Visitor(this.rule);
+
+  @override
+  void visitMethodDeclaration(MethodDeclaration node) {
+    for (var annotation in node.metadata) {
+      if (annotation.name.name == 'soloTest') {
```

Comment:
In general, a better test would be something like we have in `ElementAnnotationImpl`:
```dart
  @override
  bool get isInternal => _isPackageMetaGetter(_internalVariableName);
```
Not for `package:meta` of course, but kind of like this.
Otherwise, `soloTest` could be from some other package.
And I simply don't like using unresolved data to check anything.
That's why we put this resolution information into AST!

Reply:
Ah, cool. I'll update. I figured the chances of another soloTest to be remote but it's better practice for sure. Stand by!

Reply:
Done.